Controller/Checkout/ValidatePaymentMethod.php

```diff
@@ -102,13 +102,16 @@ protected function updateQuote($token)
     {
         $data = $this->paymentApiClient->getInfo($token);
         $billingData = $this->getFormattedAddress($data['billingContact']);
+        $shippingData = $this->getFormattedAddress($data['shippingContact']);
 
         $quote = $this->checkoutSession->getQuote();
         $quote->getBillingAddress()->addData($billingData);
-        $quote->collectTotals();
-        $this->quoteRepository->save($quote);
```

Comment:
@aleksa-andersinnovation just wanted to get your thoughts on removing the `this->quoteRepository->save` line here. When its included, it overwrites the updates to an orders shipping/billing info made in the Bread modal, so it uses the logged in users address instead of the one they changed to make the order. When I remove that save call then the address properly appears with the new value in the order system, so it seems fine to remove it, but just wanted to see if you knew more about the purpose of that than me that makes this a bad idea

Reply:
It should be checked when some shipping address discounts are activated, other than that I don't know much about why it's placed there, also should we allow changing shipping address in modal when customer is checking out from checkout flow, as they already picked shipping location there?